
# HG changeset patch
# User Sam Lantinga <slouken@libsdl.org>
# Date 1520745625 28800
# Node ID a358a111ff690a05c7c0bd04c41d5c046028595c
# Parent  6e61fc35c129578f82e909e6c8250bad74067a24
Fixed bug 4103 - SDL_cpuinfo.h provokes warnings with -Wundef on non-x86 hosts

Felix Geyer

Forwarding from https://bugs.debian.org/892087 quoting verbatim:

The SDL2 header SDL_cpuinfo.h generates gcc warnings if the program using
it compiles with the -Wundef warning. (In particular, this means that QEMU
builds using it fail on at least sparc hosts, since QEMU dev builds
use both -Wundef and -Werror.).

/usr/include/SDL2/SDL_cpuinfo.h:63:5: warning: "HAVE_IMMINTRIN_H" is not defined, evaluates to 0 [-Wundef]
 #if HAVE_IMMINTRIN_H && !defined(SDL_DISABLE_IMMINTRIN_H)

diff -r 6e61fc35c129 -r a358a111ff69 include/SDL_cpuinfo.h
--- a/include/SDL_cpuinfo.h	Sat Mar 10 21:16:14 2018 -0800
+++ b/include/SDL_cpuinfo.h	Sat Mar 10 21:20:25 2018 -0800
@@ -51,7 +51,7 @@
 #include <intrin.h>
 #else
 #ifdef __ALTIVEC__
-#if HAVE_ALTIVEC_H && !defined(__APPLE_ALTIVEC__) && !defined(SDL_DISABLE_ALTIVEC_H)
+#if defined(HAVE_ALTIVEC_H) && !defined(__APPLE_ALTIVEC__) && !defined(SDL_DISABLE_ALTIVEC_H)
 #include <altivec.h>
 #undef pixel
 #undef bool
@@ -60,7 +60,7 @@
 #if defined(__3dNOW__) && !defined(SDL_DISABLE_MM3DNOW_H)
 #include <mm3dnow.h>
 #endif
-#if HAVE_IMMINTRIN_H && !defined(SDL_DISABLE_IMMINTRIN_H)
+#if defined(HAVE_IMMINTRIN_H) && !defined(SDL_DISABLE_IMMINTRIN_H)
 #include <immintrin.h>
 #else
 #if defined(__MMX__) && !defined(SDL_DISABLE_MMINTRIN_H)

